src/main/java/org/jabref/gui/FindUnlinkedFilesDialog.java

```diff
@@ -527,6 +534,100 @@ public void stateChanged(ChangeEvent e) {
         });
     }
 
+    /**
+     * This will start the import of all file of all selected nodes in this
+     * dialogs tree view. <br>
+     * <br>
+     * The import itself will run in a seperate thread, whilst this dialog will
+     * be showing a progress bar, until the thread has finished its work. <br>
+     * <br>
+     * When the import has finished, the {@link #importFinishedHandler(java.util.List)} is
+     * invoked.
+     */
+    private void startExport() {
+
+        if (treeModel == null) {
+            return;
+        }
+        setDefaultCloseOperation(WindowConstants.DO_NOTHING_ON_CLOSE);
+
+        CheckableTreeNode root = (CheckableTreeNode) treeModel.getRoot();
+
+        final List<File> fileList = getFileListFromNode(root);
+
+        if ((fileList == null) || fileList.isEmpty()) {
+            return;
+        }
+
+        progressBarImporting.setVisible(true);
+        labelExportingInfo.setVisible(true);
+        buttonExport.setVisible(false);
+        buttonApply.setVisible(false);
+        buttonClose.setVisible(false);
+        disOrEnableDialog(false);
+
+        labelExportingInfo.setEnabled(true);
+
+        progressBarImporting.setMinimum(0);
+        progressBarImporting.setMaximum(fileList.size());
+        progressBarImporting.setValue(0);
+        progressBarImporting.setString("");
+
+        FileDialogConfiguration fileDialogConfiguration = new FileDialogConfiguration.Builder()
+                .withInitialDirectory(Globals.prefs.get(JabRefPreferences.WORKING_DIRECTORY)).build();
+        DialogService ds = new FXDialogService();
+
+        Optional<Path> exportPath = DefaultTaskExecutor
+                .runInJavaFXThread(() -> ds.showFileSaveDialog(fileDialogConfiguration));
+
+        VerifyingWriter verifyingWriter;
+        try {
+            verifyingWriter = new VerifyingWriter(Files.newOutputStream(exportPath.get()), Charset.defaultCharset());
+        } catch (Exception e) {
+            LOGGER.warn("Error while opening file.", e);
+            exportFinishedHandler();
+            return;
+        }
+
+        threadState.set(true);
+        JabRefExecutorService.INSTANCE.execute(() -> {
+            List<String> errors = new LinkedList<>();
```

Comment:
According to the codacy output, the variable errors is never used. And in 90% the cases a normal ArrayList is sufficient, a LinkedList costs more memory. The benefits of a LinkedList seldom outweigh the memory usage in comparsion to ArrayList

Reply:
I'll remove it.